StringUtil.java

```diff
@@ -1,26 +1,84 @@
 
+
 /**
  * Created by IceBreakers on 2/4/17.
  * Reverse a sentence
  * Url Parser
  * Input:
  */
 public class StringUtil {
-
+	
+	
+	
     private static String[] URLParser(String url){
         //WRITE YOUR CODE HERE..
         //parse the url and get all the get parameters and return an array of Strings
         //Input: google.com/api?param1=value1&param2=value2
         //Output: array of strings [] = { {param1:value1}, {param2:value2} }
-
+    	int ind = url.indexOf('?');
+    	String param = url.substring(ind+1);
+    	String[] paramList = param.split("&");
+    	int n = paramList.length;
+    	String[] newparamList = new String[n];
+    	for(int i=0;i<n;i++){
+    		newparamList[i] = "{";
+    		newparamList[i] = newparamList[i].concat(paramList[i]);
+    		newparamList[i] = newparamList[i].concat("}");
+    		newparamList[i] = newparamList[i].replace('=', ':');
+    	}
+    	return newparamList;
     }
+    
     private static String reverseWords(String sentence){
         //WRITE YOUR CODE HERE..
+    	if(sentence==null || sentence==""){
+    		return sentence;
+    	}
+    	String output = "";
+    	String[] strArr = sentence.split(" ");
+    	int n = strArr.length - 1;
+    	for(int i=n;i>=0;i--){
+    		output = output.concat(strArr[i]);
+    		if(i!=0){
+    		output = output.concat(" ");	
+    		}
+    	}
+    	return output;
     }
 
     private static String reverseString(String sentence){
         //WRITE YOUR CODE HERE..
+    	if(sentence==null || sentence==""){
+    		return sentence;
+    	}
+    	String output = "";
+    	String[] strArr = sentence.split(" ");
+    	int n = strArr.length - 1;
+    	for(int i=0; i<=n;i++){
+    		output = output.concat(reverse(strArr[i]));
```

Comment:
this is where you ll significantly notice the diff btw using string and stringbuilder. if you use stringbuilder its like using character array and it dynamically grows. but if you use string everytime it creates a new string since string is an immutable object.